header/testing.go

```diff
@@ -198,6 +201,45 @@ func RandBlockID(t *testing.T) types.BlockID {
 	return bid
 }
 
+// FraudMaker creates a custom ConstructFn that breaks the block at the given height.
+func FraudMaker(t *testing.T, faultHeight int64) ConstructFn {
+	log.Infow("Corrupting block...", "height", faultHeight)
+	return func(ctx context.Context,
+		b *types.Block,
+		comm *types.Commit,
+		vals *types.ValidatorSet,
+		bServ blockservice.BlockService) (*ExtendedHeader, error) {
+		eh := &ExtendedHeader{
+			RawHeader:    b.Header,
+			Commit:       comm,
+			ValidatorSet: vals,
+		}
+
+		if b.Height == faultHeight {
+			eh = CreateFraudExtHeader(t, eh, bServ)
+			return eh, nil
+		}
+
+		namespacedShares, _, _ := b.Data.ComputeShares(uint64(0))
+		extended, err := ipld.AddShares(ctx, namespacedShares.RawShares(), bServ)
+		require.NoError(t, err)
+		dah := da.NewDataAvailabilityHeader(extended)
+		eh.DAH = &dah
```

Comment:
Would be nicer to reuse the code here from real HeaderMaker, but that's optional

Reply:
Yes, sounds good